javaparser-core/src/main/java/com/github/javaparser/printer/PrettyPrintVisitor.java

```diff
@@ -712,37 +715,100 @@ public void visit(final SuperExpr n, final Void arg) {
     @Override
     public void visit(final MethodCallExpr n, final Void arg) {
         printComment(n.getComment(), arg);
+
+        // we are at the last method call of a call chain
+        // this means we do not start reindenting for alignment or we undo it
+        AtomicBoolean lastMethodInCallChain = new AtomicBoolean(true);
```

Comment:
JavaParser only promises thread safety when you don't reuse object instances, so this could be a boolean, I think.

Reply:
Nope, this is not about thread safety, but about finalness. The booleans are used inside the `n.getScope().ifPresent` lambda and for this they have to be effectively final.
An alternative is to copy the booleans into another boolean fields that then is effectively final, or using a one-element array, but I usually prefer using `AtomicBoolean` or `AtomicReference` for this as it is more object oriented unless they cause a serious performance issue.

Reply:
Oh hey, I learned a new trick :-)